[weibo]下载清晰度太低，建议尽量用PC端的地址解析

目前这微博都是通过转换成手机端上的URL来获取视频地址的，但这影响清晰度，而且手机上的解析得到的最高清晰度与电脑还是有差别，如下这个视频，手机上最高清晰度下载完文件为9.5M，而在电脑上直接打开下载则有15M大

@iawia002 为什么我这样执行会报错：
`D:\go_workspace>annie -i https://weibo.com/tv/v/abc?fid=1034:4255266544534450
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/iawia002/annie/extractors.Weibo(0xc04205aa00, 0x1d, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        C:/Users/admin/go/src/github.com/iawia002/annie/extractors/weibo.go:18 +0x59f
main.download(0xc042080040, 0x34)
        C:/Users/admin/go/src/github.com/iawia002/annie/main.go:80 +0x4f1
main.main()
        C:/Users/admin/go/src/github.com/iawia002/annie/main.go:121 +0xe7
`

--- a/extractors/weibo/weibo.go
+++ b/extractors/weibo/weibo.go
@@ -1,22 +1,82 @@
 package weibo
 
 import (
+	"fmt"
+	netURL "net/url"
 	"strings"
 
 	"github.com/iawia002/annie/downloader"
+	"github.com/iawia002/annie/parser"
 	"github.com/iawia002/annie/request"
 	"github.com/iawia002/annie/utils"
 )
 
+func downloadWeiboTV(url string) ([]downloader.Data, error) {
+	headers := map[string]string{
+		"Cookie": "SUB=_2AkMsZ8xOf8NxqwJRmP4RzGLqbo5xyQDEieKaOz2VJRMxHRl-yj83qlEotRB6B-fiobWQ5vdEoYw7bCoCdf4KyP8O3Ujq",
+	}
+	html, err := request.Get(url, url, headers)
+	if err != nil {
+		return downloader.EmptyList, err
+	}
+	doc, err := parser.GetDoc(html)
+	if err != nil {
+		return downloader.EmptyList, err
+	}
+	title := strings.TrimSpace(
+		strings.Replace(doc.Find(".info_txt").First().Text(), "\u200B", " ", -1), // Zero width space.
+	)
+	// http://f.us.sinaimg.cn/003Cddn4lx07oCX1hC0001040200hkQk0k010.mp4?label=mp4_hd&template=852x480.20&Expires=1541041515&ssig=%2BYnCmZaToS&KID=unistore,video
+	// &480=http://f.us.sinaimg.cn/003Cddn4lx07oCX1hC0001040200hkQk0k010.mp4?label=mp4_hd&template=852x480.20&Expires=1541041515&ssig=%2BYnCmZaToS&KID=unistore,video
+	// &720=http://f.us.sinaimg.cn/004cqzndlx07oCX1kMOQ01040200vyxj0k010.mp4?label=mp4_720p&template=1280x720.20&Expires=1541041515&ssig=Fdasnr1aW6&KID=unistore,video&qType=720
+	realURL, err := netURL.PathUnescape(
+		utils.MatchOneOf(html, `video-sources="fluency=(.+?)"`)[1],
+	)
+	if err != nil {
+		return downloader.EmptyList, err
+	}
+	quality := []string{"480", "720"}
+	streams := make(map[string]downloader.Stream, len(quality))
+	for _, q := range quality {
+		urlList := strings.Split(realURL, fmt.Sprintf("&%s=", q))
+		u := urlList[len(urlList)-1]
+		if !strings.HasPrefix(u, "http") {
+			continue
+		}
+		size, err := request.Size(u, url)
+		if err != nil {
+			return downloader.EmptyList, err
+		}
+		streams[q] = downloader.Stream{
+			URLs: []downloader.URL{
+				{
+					URL:  u,
+					Size: size,
+					Ext:  "mp4",
+				},
+			},
+			Size:    size,
+			Quality: q,
+		}
+	}
+	return []downloader.Data{
+		{
+			Site:    "微博 weibo.com",
+			Title:   title,
+			Type:    "video",
+			Streams: streams,
+			URL:     url,
+		},
+	}, nil
+}
+
 // Download main download function
 func Download(url string) ([]downloader.Data, error) {
 	if !strings.Contains(url, "m.weibo.cn") {
-		statusID := utils.MatchOneOf(url, `weibo\.com/tv/v/([^\?/]+)`)
-		if statusID != nil {
-			url = "https://m.weibo.cn/status/" + statusID[1]
-		} else {
-			url = strings.Replace(url, "weibo.com", "m.weibo.cn", 1)
+		if strings.Contains(url, "weibo.com/tv/v/") {
+			return downloadWeiboTV(url)
 		}
+		url = strings.Replace(url, "weibo.com", "m.weibo.cn", 1)
 	}
 	html, err := request.Get(url, url, nil)
 	if err != nil {

--- a/extractors/weibo/weibo_test.go
+++ b/extractors/weibo/weibo_test.go
@@ -46,6 +46,15 @@ func TestDownload(t *testing.T) {
 				Size:  9198410,
 			},
 		},
+		{
+			name: "weibo.com/tv test",
+			args: test.Args{
+				URL:     "https://weibo.com/tv/v/jGz6llNZ1?fid=1034:4298353237002268",
+				Title:   "做了这么一个屌炸天的视频我也不知道起什么标题好 @DRock-Art @毒液-致命守护者 @漫威影业 #绘画# #blender# #漫威#",
+				Quality: "720",
+				Size:    7520929,
+			},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {

